airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/per_partition_cursor.py

```diff
@@ -179,16 +123,16 @@ def _get_state_for_partition(self, partition: Mapping[str, Any]) -> Optional[Str
         return None
 
     @staticmethod
-    def _is_new_state(stream_state):
+    def _is_new_state(stream_state: Mapping[str, Any]) -> bool:
```

Comment:
Why not be strict on type here? We don't have a special `StreamState` type as that mapping yet?

Reply:
that's correct. We have types for `StreamSlice` and `Record`, but `StreamState` is still TBD